app/controllers/api/v1/registers_controller.rb

```diff
@@ -13,7 +13,31 @@ def index
     end
 
     items = registers.extract_by_date(props)
-    render_api({ items: items, years: registers.years },
+    sorted_items = items.sort { |a, b|  b.date <=> a.date }
+    filtered_items = sorted_items.first(20)
+
+    render_api({ items: filtered_items, years: registers.years },
+                 :ok, each_serializer: RegistersSerializer)
+  end
+
+  def fetch_on_scroll
```

Comment:
I don't think you need this action. `index` should be enough
just add appropriate scopes
the idea is if front-end sends params it filters, sorts, cuts, etc the records, but if not then it just returns the list of records.
`limit` and `offset` just do nothing if pass `nil` as a param, so nothing to do with these two.